File: backend/src/api/schemas/board.py
"""
Pydantic schemas for board API endpoints.
"""
from pydantic import BaseModel, Field, ConfigDict
from datetime import datetime
from uuid import UUID
from typing import Optional, List
from enum import Enum


class BoardStatusEnum(str, Enum):
    """Board status enumeration for API."""
    ACTIVE = "active"
    ARCHIVED = "archived"


class BoardCreateRequest(BaseModel):
    """Request schema for creating a board."""
    
    title: str = Field(
        ...,
        min_length=1,
        max_length=255,
        description="Board title (1-255 characters)",
        examples=["Моя доска"],
    )
    description: Optional[str] = Field(
        None,
        max_length=10000,
        description="Board description (optional, up to 10000 characters)",
        examples=["Описание моей доски"],
    )
    
    model_config = ConfigDict(json_schema_extra={
        "example": {
            "title": "Моя доска",
            "description": "Описание моей доски",
        }
    })


class BoardUpdateRequest(BaseModel):
    """Request schema for updating a board."""
    
    title: Optional[str] = Field(
        None,
        min_length=1,
        max_length=255,
        description="Board title (1-255 characters)",
    )
    description: Optional[str] = Field(
        None,
        max_length=10000,
        description="Board description (optional)",
    )
    
    model_config = ConfigDict(json_schema_extra={
        "example": {
            "title": "Обновлённое название",
            "description": "Новое описание",
        }
    })


class BoardResponse(BaseModel):
    """Response schema for board operations."""
    
    id: UUID
    title: str
    description: Optional[str]
    owner_id: UUID
    status: BoardStatusEnum
    created_at: datetime
    updated_at: datetime
    
    model_config = ConfigDict(from_attributes=True)


class BoardListResponse(BaseModel):
    """Response schema for boards list with pagination."""
    
    boards: List[BoardResponse]
    pagination: dict
    
    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/board.ts
/**
 * Board types for frontend application.
 */

/**
 * Board status enumeration.
 */
export const BOARD_STATUS = {
  ACTIVE: 'active',
  ARCHIVED: 'archived',
} as const

export type BoardStatus = typeof BOARD_STATUS[keyof typeof BOARD_STATUS]

/**
 * Board member roles.
 */
export const BOARD_ROLES = {
  OWNER: 'owner',
  ADMIN: 'admin',
  MEMBER: 'member',
} as const

export type BoardRole = typeof BOARD_ROLES[keyof typeof BOARD_ROLES]

/**
 * Board representation in the application.
 */
export interface Board {
  id: string // UUID
  title: string // 1-255 characters
  description?: string // 0-10000 characters (optional)
  ownerId: string // UUID
  status: BoardStatus
  createdAt: string // ISO 8601 timestamp
  updatedAt: string // ISO 8601 timestamp
}

/**
 * Board member with role and permissions.
 */
export interface BoardMember {
  id: string // UUID
  boardId: string // UUID
  userId: string // UUID
  role: BoardRole
  permissions: string[] // ['read', 'write', 'delete']
  createdAt: string // ISO 8601 timestamp
}

/**
 * Data transfer object for creating a new board.
 */
export interface CreateBoardDTO {
  title: string
  description?: string
}

/**
 * Data transfer object for updating a board.
 */
export interface UpdateBoardDTO {
  title?: string
  description?: string
}

/**
 * Board list response with pagination.
 */
export interface BoardListResponse {
  boards: Board[]
  pagination: {
    page: number
    limit: number
    total: number
    total_pages: number
  }
}

/**
 * Validation limits for board fields.
 */
export const BOARD_LIMITS = {
  TITLE_MIN: 1,
  TITLE_MAX: 255,
  DESCRIPTION_MAX: 10000,
} as const
